### PR TITLE
Add go licenses

### PR DESCRIPTION
**Pull request by mbargull**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This gathers licenses of vendored packages and records them in a `thirdparty-licenses.txt` file.

### CODE REVIEW

File: recipe/build.sh [hunk 1]
```diff
@@ -14,3 +14,55 @@ case $(uname -s) in
...
+
+--------------------------------------------------------------------------------
+${2#${1%/}/}
+================================================================================
+EOF
+cat "${2}"
+' -- "${acc_dir}" >> "${output}"
+  rm -r "${acc_dir}"
+}
+
```
Comment (mbargull) [inline recipe/build.sh:57]: Ideally, this `gather_licenses` function will be replaced with just `go-licenses save` calls and we let `conda-build` gather the created files then. Currently, `conda-build` only support single files under `license_file`, but once the `conda-build` issue mentioned in the comment above is resolved, I'll clean this up.

**Comment by chrisburr** [comment]
Thanks for this @mbargull! For the aarch64 and ppc64le I've added `go-licenses` to the arch migration (https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/926).

**Comment by mbargull** [comment]
Perfect, thank you!

**Review by chrisburr — APPROVED** [review]
Looks like linux-64 is actually failing but feel free to merge this yourself once the issues are fixed.

File: recipe/build.sh [hunk 2]
```diff
@@ -39,6 +39,7 @@ go-licenses csv ${module}
 EOF
     go-licenses csv "${module}" | sort >> "${output}"
     go-licenses save "${module}" --save_path="${tmp_dir}"
+    chmod -R +rw "${acc_dir}"
```
Comment (mbargull) [inline recipe/build.sh:42]: Without it I got
```
cp: cannot create regular file `/home/conda/feedstock_root/build_artifacts/cni-plugins_1604485827948/work/gather-licenses-acc/github.com/containernetworking/cni/pkg/LICENSE': Permission denied
```
but I have no clue why.
Let's see if this `chmod` changes anything..

Comment (mbargull) [reply recipe/build.sh]: That's really strange. But `chmod` made it work.
I don't get this with a local (also container-based) build. Maybe `umask` or something is different on CI? I'm confused.

**Comment by mbargull** [comment]
@conda-forge-admin, please restart ci

**Comment by chrisburr** [comment]
Thanks @mbargull! :cake:

**Pull request merged**